server/scriptController.js

```diff
@@ -20,7 +22,102 @@ const scriptController = {
       .catch((error) => {
         return next({
           log: `error: ${error} occured when getting script titles`,
-          message: 'error in getScriptTitles in scriptController' ,
+          message: 'error in getScriptTitles in scriptController',
+        });
+      });
+  },
+  saveScript: (req, res, next) => {
+    const MAXFILESIZE = 50 * 1024 * 1024; //50MB
+
+    // configure multer storage to upload files to the path from the .env file
+    const storage = multer.diskStorage({
+      destination: function (req, file, cb) {
+        cb(null, process.env.UPLOADPATH);
+      },
+      filename: function (req, file, cb) {
+        cb(null, Date.now() + file.originalname);
+      },
+    });
+    // limit file uploads to less than 50MB
+    const limits = { fileSize: MAXFILESIZE };
+
+    const upload = multer({ storage, limits });
+    const scriptUpload = upload.single('newScript');
+
+    // actually upload the script
+    scriptUpload(req, res, next, (error) => {
+      if (error)
+        return next({
+          log: `error: ${error} occured when uploading the script`,
+          message: 'error when uploading the script',
+        });
+      else {
+        return next();
+      }
+    });
+  },
+  parseScript: (req, res, next) => {
+    // run the script through playData and save the results to the database
+    const path = process.env.UPLOADPATH + req.file.filename;
+    const scriptData = parseScript(path);
+    playData[scriptData.title] = scriptData;
+    const characterObjects = Object.values(scriptData.characterObjs);
+
+    // check if the script already exists
+    db.query('SELECT * FROM scripts WHERE title=$1', [scriptData.title])
```

Comment:
I often see some tiny bit of data used in the select instead of * when all we care about is whether the select found any rows, instead of the content of those rows. This requires sending less data back from the db.  
Ex:
```sql
select 'found' from scripts where title = $1
```

Reply:
That's sleek!